File: nl.naturalis.nba.client.php/model/SpecimenIdentification.php
<?php

namespace nl\naturalis\nba\client\php\model;

use nl\naturalis\nba\client\php\model\TaxonomicIdentification;

class SpecimenIdentification extends TaxonomicIdentification {

	private $preferred;
	private $verificationStatus;
	private $rockType;
	private $associatedFossilAssemblage;
	private $rockMineralUsage;
	private $associatedMineralName;
	private $remarks;

	public function __construct()
	{
		parent::__construct();
	}

	public function functionisPreferred()
	{
		return $this->preferred;
	}


	public function functionsetPreferred(bool $preferred)
	{
		$this->preferred = $preferred;
	}


	public function functiongetVerificationStatus()
	{
		return $this->verificationStatus;
	}


	public function functionsetVerificationStatus(string $verificationStatus)
	{
		$this->verificationStatus = $verificationStatus;
	}


	public function functiongetRockType()
	{
		return $this->rockType;
	}


	public function functionsetRockType(string $rockType)
	{
		$this->rockType = $rockType;
	}


	public function functiongetAssociatedFossilAssemblage()
	{
		return $this->associatedFossilAssemblage;
	}


	public function functionsetAssociatedFossilAssemblage(string $associatedFossilAssemblage)
	{
		$this->associatedFossilAssemblage = $associatedFossilAssemblage;
	}


	public function functiongetRockMineralUsage()
	{
		return $this->rockMineralUsage;
	}


	public function functionsetRockMineralUsage(string $rockMineralUsage)
	{
		$this->rockMineralUsage = $rockMineralUsage;
	}


	public function functiongetAssociatedMineralName()
	{
		return $this->associatedMineralName;
	}


	public function functionsetAssociatedMineralName(string $associatedMineralName)
	{
		$this->associatedMineralName = $associatedMineralName;
	}


	public function functiongetRemarks()
	{
		return $this->remarks;
	}


	public function functionsetRemarks(string $remarks)
	{
		$this->remarks = $remarks;
	}


}

File: nl.naturalis.nba.api/src/main/java/nl/naturalis/nba/api/model/SpecimenIdentification.java
package nl.naturalis.nba.api.model;

import static nl.naturalis.nba.api.annotations.Analyzer.CASE_INSENSITIVE;
import static nl.naturalis.nba.api.annotations.Analyzer.DEFAULT;
import static nl.naturalis.nba.api.annotations.Analyzer.LIKE;

import nl.naturalis.nba.api.annotations.Analyzers;

public class SpecimenIdentification extends TaxonomicIdentification {

	private boolean preferred;
	private String verificationStatus;
	private String rockType;
	private String associatedFossilAssemblage;
	private String rockMineralUsage;
	private String associatedMineralName;
	@Analyzers({ CASE_INSENSITIVE, DEFAULT, LIKE })
	private String remarks;

	public boolean isPreferred()
	{
		return preferred;
	}

	public void setPreferred(boolean preferred)
	{
		this.preferred = preferred;
	}

	public String getVerificationStatus()
	{
		return verificationStatus;
	}

	public void setVerificationStatus(String verificationStatus)
	{
		this.verificationStatus = verificationStatus;
	}

	public String getRockType()
	{
		return rockType;
	}

	public void setRockType(String rockType)
	{
		this.rockType = rockType;
	}

	public String getAssociatedFossilAssemblage()
	{
		return associatedFossilAssemblage;
	}

	public void setAssociatedFossilAssemblage(String associatedFossilAssemblage)
	{
		this.associatedFossilAssemblage = associatedFossilAssemblage;
	}

	public String getRockMineralUsage()
	{
		return rockMineralUsage;
	}

	public void setRockMineralUsage(String rockMineralUsage)
	{
		this.rockMineralUsage = rockMineralUsage;
	}

	public String getAssociatedMineralName()
	{
		return associatedMineralName;
	}

	public void setAssociatedMineralName(String associatedMineralName)
	{
		this.associatedMineralName = associatedMineralName;
	}

	public String getRemarks()
	{
		return remarks;
	}

	public void setRemarks(String remarks)
	{
		this.remarks = remarks;
	}

}
